src/zuuid.c

```diff
@@ -116,38 +118,40 @@ zuuid_set (zuuid_t *self, byte *source)
         self->str [byte_nbr * 2 + 1] = hex_char [val & 15];
     }
     self->str [ZUUID_LEN * 2] = 0;
+    free (self->str_canonical);
 }
 
 
 //  -----------------------------------------------------------------
-//  Set UUID to new supplied string value skipping '-' and '{' '}' optional delimiters;
-//  return 0 if OK, else returns -1.
+//  Set UUID to new supplied string value skipping '-' and '{' '}'
+//  optional delimiters. Return 0 if OK, else returns -1.
 
 int
 zuuid_set_str (zuuid_t *self, const char *source)
 {
     assert (self);
     assert (source);
-    int str_len = strlen (source);
-    assert (str_len == ZUUID_STR_LEN || str_len == ZUUID_STR_LEN + 4 || str_len == ZUUID_STR_LEN + 6);
 
-    int byte_nbr;
-    const char *p = source;
-    for (byte_nbr = 0; byte_nbr < ZUUID_LEN; byte_nbr++) {
-        while (str_len > 0 && (*p == '-' || *p == '{' || *p == '}')) {
-            p++;
-            str_len--;
+    uint byte_nbr = 0;
+    while (*source) {
+        if (*source == '-' || *source == '{' || *source == '}')
+            source++;
+        else {
+            //  Get two hex digits
+            uint value;
+            if (sscanf (source, "%02x", &value) != 1)
+                return -1;
+            if (byte_nbr < ZUUID_LEN) {
+                self->uuid [byte_nbr] = (byte) value;
+                self->str [byte_nbr * 2 + 0] = *source++;
+                self->str [byte_nbr * 2 + 1] = *source++;
+                byte_nbr++;
+            }
+            else
+                return -1;
         }
-        if (str_len <= 0 && byte_nbr < ZUUID_LEN)
-            return -1;
-        uint value;
-        if (sscanf (p, "%02x", &value) != 1)
-            return -1;
-        self->uuid [byte_nbr] = (byte)value;
-        self->str[byte_nbr * 2] = *p++;
-        self->str[byte_nbr * 2 + 1] = *p++;
-        str_len -= 2;
     }
+    free (self->str_canonical);
```

Comment:
OK, set to null and return 'const' in both zuuid_str and
zuuid_str_canonical.

On Fri, Mar 27, 2015 at 11:17 PM, twhittock <notifications@github.com>
wrote:

> In src/zuuid.c
> <https://github.com/zeromq/czmq/pull/970#discussion_r27335107>:
>
> >      }
> > +    free (self->str_canonical);
>
> should set null here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/czmq/pull/970/files#r27335107>.
>